src/netcdf_scm/masks.py

```
@@ -0,0 +1,309 @@
+        Any optional arguments to be passed to the MaskFunc's during evaluation.
+        Possible parameters include:
+
+            sftlf_cube : ScmCube
+
+            land_mask_threshold : float default: 50.
+    """
+
+    def __init__(self, cube, **kwargs):
+        self.cube = cube
+        self._masks = {}
+        self.kwargs = kwargs
+
+    def get_mask(self, mask_name):
+        """
+        Get a single mask
+
+        If the mask has previously been calculated the precalculated result is returned from the cache. Otherwise the appropriate
+        MaskFunc is called with any kwargs specified in the constructor.
+
+        Parameters
+        ----------
+        mask_name : str
+
+        Raises
+        ------
+        InvalidMask
+            If the requested mask cannot be found or evaluated.
+
+        Returns
+        -------
+        ndarray[bool]
+            Any True values should be masked out and excluded from any further calculation.
+        """
+        try:
+            return self._masks[mask_name]
+        except KeyError:
+            try:
+                mask_func = MASKS[mask_name]
+                mask = mask_func(self, self.cube, **self.kwargs)
+                self._masks[mask_name] = mask
+            except KeyError:
+                raise InvalidMask('Unknown mask: {}'.format(mask_name))
+            except InvalidMask:
+                raise
+
+        return mask
+
+    def get_masks(self, mask_names):
+        """
+        Get arbitary masks
+
+        Parameters
+        ----------
+        mask_names: list of str
+            The masks to calculate
+
+        Returns
+        -------
+        dict of ndarrays of bool
+
+        The result only contains valid masks. Any invalid masks are dropped.
+        """
+        masks = {}
+        for name in mask_names:
+            try:
+                mask = self.get_mask(name)
+                masks[name] = mask
+            except InvalidMask:
+                pass
```

do we want an error/warning here explaining what happened

---

The exception message won't be all that helpful here as it won't always be due to the root cause of it failing in the case of cascading failures.
I.e missing land mask > can't make ocean mask > can't make NH|Ocean mask

---

can we make it a warning, "Failed to get mask for 'NH|Ocean'" so users at least know? This is a silent fail which I'm uncomfortable with.